src/foam/u2/view/ArrayView.js

```diff
@@ -35,7 +39,11 @@ foam.CLASS({
         return mode === foam.u2.DisplayMode.RW;
       },
       code: function() {
-        this.data[this.data.length] = '';
+        var newItem = this.defaultNewItem;
+        if ( foam.core.FObject.isInstance(newItem) ) {
```

Comment:
Should add

```JavaScript
    'foam.core.FObject',
```

to the `requires` array and change this line to

```suggestion
        if ( this.FObject.isInstance(newItem) ) {
```

Reply:
Updated this